tests/Unit/Pypp/CheckWithRandomValues.hpp

```diff
@@ -475,13 +592,15 @@ void check_with_random_values(
                                      tmpl::size<argument_types>>,
                       TestHelpers_detail::RemoveNotNull<tmpl::_1>>;
 
-  static_assert(number_of_not_null::value != 0,
-                "You must return at least one argument by gsl::not_null when"
-                "passing the python function names as a vector<string>. If "
-                "your function returns by value do not pass the function name "
-                "as a vector<string> but just a string.");
   static_assert(
-      cpp17::is_same_v<typename f_info::return_type, void>,
+      number_of_not_null::value != 0 or tt::is_a_v<tmpl::list, TagsList>,
+      "You must return at least one argument by gsl::not_null when"
+      "passing the python function names as a vector<string>. If "
+      "your function returns by value do not pass the function name "
+      "as a vector<string> but just a string.");
```

Comment:
this should probably mention the `or` part of the check, same with the one below